src/yang/ietf-access-control-list.yang

```diff
@@ -443,21 +443,61 @@ module ietf-access-control-list {
             choice l4 {
               container tcp {
                 if-feature match-on-tcp;
-                uses packet-fields:acl-tcp-header-fields;
+                uses pf:acl-tcp-header-fields;
+                choice source-port {
+                  container source-port-range-or-operator {
+                    uses pf:port-range-or-operator;
+                    description
+                      "Source port definition.";
+                  }
                   description
-                    "Rule set that matches TCP headers.";
+                    "Choice of specifying the source port or referring to
+                     a group of source ports.";
+                }
+                choice destination-port {
+                  container destination-port-range-or-operator {
+                    uses pf:port-range-or-operator;
+                    description
+                      "Destination port definition.";
+                  }
+                  description
+                    "Choice of specifying a destination port or referring
+                     to a group of destination ports.";
+                }
+                description
+                  "Rule set that matches TCP headers.";
               }
 
               container udp {
                 if-feature match-on-udp;
-                uses packet-fields:acl-udp-header-fields;
+                uses pf:acl-udp-header-fields;
+                choice source-port {
+                  container source-port-range-or-operator {
```

Comment:
Isn't this an extra container added in? Why?

Reply:
This is future proofing it for when object groups are added as the other choice.

Reply:
I get why you are adding the choice but why add "container source-port-range-or-operator"?

Reply:
I do not understand. How else would you enclose the following uses statement as a case within the choice statement? 

Maybe if you could suggest an alternative, it would be easier to understand the point you are trying to make??

Reply:
I know it is probably too late for you to respond, and I have to go ahead submit an update to the draft. I will still have an opportunity to make changes, specially if the change is editorial.

Reply:
The normal way of enclosing it is using `case`. Please read https://tools.ietf.org/html/rfc7950#page-27

Having a single leaf or a container in a `choice` statement can be used as a short-hand and thus omitting the `case` statement.

Having that container taints the data tree. Now there will be a useless source-port-range-or-operator container in the configuration since you are using a `container` rather than `case`. Someone using a CLI will actually have to type in the configuration `source-port-range-or-operator`. It does not bear meaning to the final configuration, only as a useless schema node to sort things out, which is why I don't think it should be in there. Replace with `case`.

Reply:
Okay, I see now in the master branch what's there. You already have the container named source-port-range-or-operator. It should just be "source-port".

Reply:
I so wish you had just used the changes from my branch for the original patches we applied instead of rewriting everything based off email threads and we would have avoided so many issues.

Reply:
I have now opened a PR to your repo. You just have to merge that one first, then this PR can pull in that change into netmod-wg/acl-model too.